src/PeriodEnd/ProviderAdjustments/SFA.DAS.Payments.Calc.ProviderAdjustments/ProviderAdjustmentsProcessor.cs

```diff
@@ -36,29 +42,27 @@ public virtual void Process()
         {
             _logger.Info("Started the Provider Adjustments Processor.");
 
-            var collectionPeriod = ReturnValidGetCurrentCollectionPeriodQueryResponseOrThrow();
-            var providersQueryResponse = ReturnValidGetProvidersQueryResponseOrThrow();
+            var previousPayments = _adjustmentRepository.GetPreviousProviderAdjustments();
+            var earnings = _adjustmentRepository.GetCurrentProviderAdjustments();
 
-            if (providersQueryResponse.HasAnyItems())
-            {
-                foreach (var provider in providersQueryResponse.Items)
-                {
-                    _logger.Info($"Processing adjustments for provider with ukprn {provider.Ukprn}.");
+            var payments = _providerPayments.CalculatePayments(previousPayments.ToList(), earnings.ToList());
 
-                    var currentAdjustmentsResponse = ReturnValidGetCurrentAdjustmentsQueryResponseOrThrow(provider.Ukprn);
-                    var previousPaymentsResponse = ReturnValidGetPreviousAdjustmentsQueryResponseOrThrow(provider.Ukprn);
+            PopulateCollectionPeriod(payments);
+            _paymentRepository.AddProviderAdjustments(payments.ToArray());
 
-                    ProcessAdjustmentsForProvider(provider, currentAdjustmentsResponse.Items, previousPaymentsResponse.Items, collectionPeriod.Period);
+            _logger.Info("Finished the Provider Adjustments Processor.");
+        }
 
-                    _logger.Info($"Finished processing adjustments for provider with ukprn {provider.Ukprn}.");
-                }
-            }
-            else
+        private void PopulateCollectionPeriod(IEnumerable<PaymentEntity> payments)
```

Comment:
Does this method belong to the CalculatePayments Class.

Reply:
It's not really doing anything around calculations and is more infrastructure related